dts/bindings/gpio/adi,max14906-gpio.yaml

```diff
@@ -0,0 +1,87 @@
+# Copyright (c) 2024 Analog Devices Inc.
+# Copyright (c) 2024 BayLibre SAS
+# SPDX-License-Identifier: Apache-2.0
+
+description: ADI MAX14906 quad industrial Input/Output with advanced diagnostics
+
+compatible: "adi,max14906-gpio"
+
+properties:
+  "#gpio-cells":
+    const: 2
+  ngpios:
+    type: int
+    required: true
+    const: 4
+    description: Number of gpios supported
+  drdy-gpios:
+    description: Ready pin which show when chip is ready
+    type: phandle-array
+  fault-gpios:
+    description: |
+      Fault pin indicates when there is Fault state in either FAULT1 or FAULT2
+      bothe of which are cleaned on read once problem is not persistent
+    type: phandle-array
+  sync-gpios:
+    description: |
+      Latch the data so it could be read (partially duplicate CS)
+    type: phandle-array
+  en-gpios:
+    description: |
+      Latch the data so it could be read (partially duplicate CS)
+    type: phandle-array
+  crc-en:
+    description: to notify driver if crc pin is enabled
+    type: boolean
+  spi-addr:
+    type: int
+    default: 0
```

Comment:
Need to explain in the description why the default value was selected
https://docs.zephyrproject.org/latest/build/dts/bindings-upstream.html#dt-bindings-default-rules

Same for the following properties below

Reply:
Thanks , explanation is added

Reply:
This also needs to be done for the properties below

Reply:
Added default values are from documentation - default value per register

Reply:
Actually, why is this property needed? It's redundant to the `reg` property

Reply:
I think using `reg` for that purpose will be kind of confusing.
`spi-addr` is something specific to MAX14906 and MAX14916.